addons/knowledge/static/src/js/knowledge_frontend.js

```diff
@@ -86,4 +88,25 @@ publicWidget.registry.KnowledgeWidget = publicWidget.Widget.extend(KnowledgeTree
             });
         });
     },
+
+    /**
+     * Scroll through the view to display the matching heading.
+     * Adds a small highlight in/out animation using a class.
+     *
+     * @param {Event} event
+     */
+    _onTocLinkClick: function (event) {
+        event.preventDefault();
+        const headingIndex = parseInt(event.target.getAttribute('data-oe-nodeid'));
+        const targetHeading = fetchRelevantHeadings(this.$el[0])[headingIndex];
+        if (targetHeading){
+            targetHeading.scrollIntoView({
+                behavior: 'smooth',
+            });
+            targetHeading.setAttribute('highlight', true);
+            setTimeout(() => {
+                targetHeading.removeAttribute('highlight');
+            }, 2000);
+        }
+    },
```

Comment:
Can't we put this in a common assets file to avoid code duplication ?

Reply:
I thought about it but the code is "slightly different".
We don't use the same target for "fetchRelevantHeadings" and the backend version updates your ToC if it doesn't find the link.

That makes me unsure it's worth avoiding duplication as we will have custom parameters and/or custom overrides.